### PR TITLE
Double64 gear solver

### PR DESCRIPTION
**Pull request by planetarymike**
Made function argument types more generic, added custom gear solver using Double64 type.

### CODE REVIEW

File: converge_new_file.jl [hunk 1]
```diff
@@ -549,27 +551,245 @@ function evolve_atmosphere(atm_init::Dict{Symbol, Array{ftype_ncur, 1}}, log_t_s
...
+    tlower, tup, tdown, tupper = update_transport_coefficients(transport_species, updated_ncur_all, activellsp, Tn, Tp, D_arr, Hs_dict, speciesbclist,
+                                                               all_species, neutral_species, transport_species, molmass, n_alt_index, polarizability, alt, num_layers, dz, Tprof_for_diffusion)
+
+    return (ratefn(n, n_short_updated, inactive, activellsp, activeslsp, inactivesp, Jrates, Tn, Ti, Te, tup, tdown, tlower, tupper),
+            chemJmat(n, n_short, n_inactive, activellsp, activeslsp, inactivesp, Jrates, Tn, Ti, Te, tup, tdown, tlower, tupper; dt=t) )# dt unused in chemJmat
+end
+
+using IterativeSolvers, IncompleteLU
+
+function solve_sparse(A, b)
```
Comment (emcangi) [inline converge_new_file.jl:643]: Could you add a docstring?

File: converge_new_file.jl [hunk 2]
```diff
@@ -549,27 +551,245 @@ function evolve_atmosphere(atm_init::Dict{Symbol, Array{ftype_ncur, 1}}, log_t_s
...
+        # restrict values to be positive definite
+        nthis[nthis .< 0.] .= 0.
+        
+        # check for convergence
+        check_n_abserr = (nthis .<= abstol) .|| (nold .<= abstol) .|| (abs.(nthis - nold) .<= abstol)
+        n_relerr = abs.(nthis-nold)./nold
+        println("max(n_relerr) = $(max(n_relerr[.!check_n_abserr]...))")
+        check_n_relerr = n_relerr .< reltol
+
+        # check_f_abserr = (abs.(fval) .<= f_abstol)
```
Comment (emcangi) [inline converge_new_file.jl:702]: should I just erase this block?

**Review by emcangi — APPROVED** [review]

**Comment by planetarymike** [comment]
Hi Eryn,

Do you want me to submit another pull request with a docstring and some
more info about the commented out code?

Mike

**Comment by emcangi** [comment]
No worries, I think it's fine for now.

**Comment by planetarymike** [comment]
With respect to the commented out block in next_timestep, that code was
intended to check if the values for the next timestep actually satisfy the
update equation; right now this code only checks if the updates to n_end
get sufficiently small, not whether the equation is actually satisfied.
This was an issue with the original code too, and should be fixed before
doing any serious time-variable modeling, but I had trouble getting to
large timesteps with this check enabled.

**Pull request merged**